### PR TITLE
Add `throwHttpErrors` option

### LINKED ISSUE #9: `throwHttpErrors` option
It's always true. Let's make it an option so we can turn that off.

### PR DESCRIPTION
**Pull request by killynathan**
Fixes #9 

adds `throwHttpErrors` option defaulted to `true` along with updated docs and tests.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -98,12 +98,16 @@ class Ky {
...
+			throwHttpErrors: true,
 			...options
 		};
 
 		this._timeout = this._options.timeout;
 		delete this._options.timeout;
 
+		this._throwHttpErrors = this._options.throwHttpErrors;
+		delete this._options.throwHttpErrors;
+
```
Comment (szmarczak) [inline index.js]: Remove these lines.

File: index.js [hunk 2]
```diff
@@ -153,7 +157,9 @@ class Ky {
 					return retry();
 				}
 
-				throw error;
+				if (this._throwHttpErrors) {
```
Comment (szmarczak) [inline index.js]:
```
if (this._options.throwHttpErrors) {
```

**Comment by szmarczak** [comment]
I'd follow https://github.com/sindresorhus/ky/issues/15

File: index.js (hunk 1 above)
Comment (killynathan) [reply index.js]: Got it. I thought I should delete it because like timeout, it's not something we want to pass to fetch.

**Comment by szmarczak** [comment]
LGTM.

**Pull request merged**